framework/src/org/checkerframework/framework/type/QualifierHierarchy.java

```diff
@@ -114,31 +114,41 @@ public abstract boolean isSubtype(
     public abstract AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2);
 
     /**
-     * Returns whether or not this hierarchy implements {@link #widenUpperBound(AnnotationMirror,
-     * AnnotationMirror)}
+     * Returns the number of iterations dataflow should preform before {@link
+     * #widenedUpperBound(AnnotationMirror, AnnotationMirror)} is called or -1 if it should never be
+     * called.
      *
-     * @return whether or not this hierarchy implements {@link #widenUpperBound(AnnotationMirror,
-     *     AnnotationMirror)}
+     * <p>Subclasses overriding this method should return some positive number.
```

Comment:
The specification doesn't state that -1 is the default value. So overriding the method with -1 to turn off widening would seem fine.